crates/re_types_builder/src/codegen/cpp/mod.rs

```diff
@@ -595,6 +609,458 @@ fn arrow_data_type_method(
     }
 }
 
+fn new_arrow_array_builder_method(datatype: &DataType, cpp_includes: &mut Includes) -> Method {
+    let arrow_builder_type = arrow_array_builder_type(datatype);
+    let arrow_builder_type = format_ident!("{arrow_builder_type}");
+    let builder_instantiation =
+        quote_arrow_array_builder_type_instantiation(datatype, cpp_includes, true);
+
+    Method {
+        docs: "Creates a new array builder with an array of this type.".into(),
+        declaration: MethodDeclaration {
+            is_static: true,
+            return_type: quote! { arrow::Result<std::shared_ptr<arrow::#arrow_builder_type>> },
+            name_and_parameters: quote!(new_arrow_array_builder(arrow::MemoryPool * memory_pool)),
+        },
+        definition_body: quote! {
+            if (!memory_pool) {
+                return arrow::Status::Invalid("Memory pool is null.");
+            }
+            #NEWLINE_TOKEN
+            #NEWLINE_TOKEN
+            return arrow::Result(#builder_instantiation);
+        },
+        inline: false,
+    }
+}
+
+fn fill_arrow_array_builder_method(
+    datatype: &DataType,
+    obj: &Object,
+    pascal_case_ident: &Ident,
+    cpp_includes: &mut Includes,
+) -> Method {
+    let DataType::Extension(_fqname, logical_datatype, _metadata) = datatype else {
+        panic!("Can only generate arrow serialization code for extension types.");
```

Comment:
always include the offending object/field's `fqname` into these build-time errors!